app/components/views/ProposalDetails/ChooseVoteOption.js

```diff
@@ -0,0 +1,125 @@
+import { FormattedMessage as T } from "react-intl";
+import { PassphraseModalButton } from "buttons";
+import { fetchMachine } from "stateMachines/FetchStateMachine";
+import { useMachine } from "@xstate/react";
+import { StakeyBounceXs } from "indicators";
+import { useDispatch } from "react-redux";
+import { useState } from "react";
+import { ProposalError } from "./helpers";
+import * as gov from "actions/GovernanceActions";
+
+const VoteOption = ({ value, description, onClick, checked }) => (
+  <div className="proposal-vote-option">
+    <input className={value} type="radio" id={value} name="proposalVoteChoice"
+      readOnly={!onClick} onChange={onClick}
+      value={value}
+      checked ={checked}
+    />
+    <label className={"radio-label " + value} htmlFor={value}/>{description}
+  </div>
+);
+
+function UpdateVoteChoiceModalButton({ onSubmit, newVoteChoice, eligibleTicketCount }) {
+  return (
+    <PassphraseModalButton
+      modalTitle={
+        <>
+          <T id="proposals.updateVoteChoiceModal.title" m="Confirm Your Vote" />
+          <div className="proposal-vote-confirmation">
+            <div className={newVoteChoice+"-proposal"}/>
+            {newVoteChoice}
+          </div>
+        </>
+      }
+      modalDescription={
+        <T
+          id="proposalDetails.votingInfo.eligibleCount"
+          m="You have {count, plural, one {one ticket} other {# tickets}} eligible for voting"
+          values={{ count: eligibleTicketCount }}
+        />
+      }
+      disabled={!newVoteChoice}
+      onSubmit={onSubmit}
+      buttonLabel={<T id="proposals.updateVoteChoiceModal.btnLabel" m="Cast Vote" />}
+    />
+  );}
+
+function getError(error) {
+  if (!error) return;
+  if (typeof error === "string") return error;
+  if (typeof error === "object") {
+    if (error.message) return error.message;
+    return JSON.stringify(error);
+  }
+}
+
+function ChooseVoteOption({
```

Comment:
Here, as it is a component, I'd rather let it as function. Otherwise, as an arrow function it does not show at the stack trace, when erroring, which make debug harder.

Reply:
I see, makes sense